### ISSUE TITLE
[BUG]: Checkbox with role=switch fails with/without aria-checked

### ISSUE THREAD

**Issue by scottnath** [IBMa/equal-access#842]
### Project

a11y checker

### Description

A checkbox with `role="switch"` and `aria-checked` returns an `The ARIA attribute 'aria-checked' is not valid for the element <input> with ARIA role 'switch'` error. [see aat error page](https://able.ibm.com/rules/archives/latest/doc/en-US/aria_semantics_attribute.html#%7B%22message%22%3A%22The%20ARIA%20attribute%20'aria-checked'%20is%20not%20valid%20for%20the%20element%20%3Cinput%3E%20with%20ARIA%20role%20'switch'%22%2C%22snippet%22%3A%22%3Cinput%20aria-label%3D%5C%22Disabled%20heart%5C%22%20aria-labelledby%3D%5C%22label-17x1i%5C%22%20disabled%3D%5C%22%5C%22%20aria-checked%3D%5C%22false%5C%22%20role%3D%5C%22switch%5C%22%20id%3D%5C%2217x1i%5C%22%20type%3D%5C%22checkbox%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_1%22%2C%22ruleId%22%3A%22aria_semantics_attribute%22%7D)

but if we remove `aria-checked`, then we get the error `An element with ARIA role 'switch' does not have the required ARIA attribute(s): 'aria-checked'` [see aat error](https://able.ibm.com/rules/archives/latest/doc/en-US/Rpt_Aria_RequiredProperties.html#%7B%22message%22%3A%22An%20element%20with%20ARIA%20role%20'switch'%20does%20not%20have%20the%20required%20ARIA%20attribute(s)%3A%20'aria-checked'%22%2C%22snippet%22%3A%22%3Cinput%20aria-label%3D%5C%22save%5C%22%20aria-labelledby%3D%5C%22label-tpwy%5C%22%20role%3D%5C%22switch%5C%22%20id%3D%5C%22tpwy%5C%22%20type%3D%5C%22checkbox%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_1%22%2C%22ruleId%22%3A%22Rpt_Aria_RequiredProperties%22%7D)

### Steps to reproduce

Changes can be seen in this ibm-internal pull request
https://github.ibm.com/w3ds/w3ds/pull/1937

**Comment by nmerget** [comment]
It's still an [issue](https://able.ibm.com/rules/archives/2024.05.09/doc/en-US/aria_attribute_valid.html#%7B%22message%22%3A%22The%20ARIA%20attributes%20%5C%22aria-checked%5C%22%20are%20not%20valid%20for%20the%20element%20%3Cinput%3E%20with%20ARIA%20role%20%5C%22switch%5C%22%22%2C%22snippet%22%3A%22%3Cinput%20aria-invalid%3D%5C%22false%5C%22%20id%3D%5C%22switch-a66d6335-2bcd-4aa3-99a0-4ce1658ec38e%5C%22%20aria-checked%3D%5C%22false%5C%22%20role%3D%5C%22switch%5C%22%20type%3D%5C%22checkbox%5C%22%3E%22%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_invalid_role_attr%22%2C%22ruleId%22%3A%22aria_attribute_valid%22%2C%22msgArgs%22%3A%5B%22aria-checked%22%2C%22input%22%2C%22switch%22%5D%7D) 😢

It should come from [here](https://github.com/IBMa/equal-access/blob/f048e794d71246220789e6f76fbcc20135f79179/accessibility-checker-engine/src/v4/rules/aria_semantics.ts#L104). But I couldn't find the issue...

**Comment by marcjohlic** [comment]
Moving this into v6.2 Release Backlog.  On Triage Shunguo pointed out that Marnie pulled this one into New Issues on the Project board.
